variables.tf

```diff
@@ -3,139 +3,169 @@
 # ------------------------------------------------------------------------------
 
 variable "apply_immediately" {
-  type        = "string"
   description = "Specifies whether any database modifications are applied immediately, or during the next maintenance window"
+  type        = string
   default     = "false"
 }
 
 variable "name_prefix" {
   description = "A prefix used for naming resources."
+  type        = string
+  default     = null
```

Comment:
This did not use to have a default value, so I don't think it should be added here.

Reply:
I had to check the documentation for this but imho I thought null will enforce the value to be defined by a user of the module the same way not filling the default value did the same in past versions.

I have tested that by not giving `name_prefix` any value and ended up with plenty of:
```
Error: Invalid template interpolation value

  on terraform-aws-rds-instance/main.tf line 46, in resource "aws_db_subnet_group" "main":
  46:   name        = "${var.name_prefix}-subnet-group"
    |----------------
    | var.name_prefix is null

The expression result is null. Cannot include a null value in a string
template.
```

Reply:
Setting `default` means that the variable becomes optional. I'm assuming in the example above, you did not supply `name_prefix` as input to the module, and since you've set the default to be `null`, you get that error message.

If you remove the default value for `name_prefix`, the module will complain about missing a required input instead, which is how it used to work, and what we want :)